integration-tests/tests/src/tests/observable.ts

```diff
@@ -1379,304 +1387,571 @@ describe("Observable", () => {
 
       static schema: ObjectSchema = {
         name: "ObjectWithMixed",
         properties: {
-          mixedValue: "mixed",
+          mixed: "mixed",
         },
       };
     }
 
-    type CollectionsInMixedContext = {
+    type CollectionsInMixedContext = RealmContext & {
       objectWithList: Realm.Object<ObjectWithMixed> & ObjectWithMixed;
       objectWithDictionary: Realm.Object<ObjectWithMixed> & ObjectWithMixed;
-      list: Realm.List<any>;
-      dictionary: Realm.Dictionary<any>;
-    } & RealmContext;
+    };
 
     openRealmBeforeEach({ schema: [ObjectWithMixed] });
 
     beforeEach(function (this: CollectionsInMixedContext) {
-      this.objectWithList = this.realm.write(() => {
-        return this.realm.create(ObjectWithMixed, { mixedValue: [] });
-      });
-      this.list = this.objectWithList.mixedValue as Realm.List<any>;
-      expect(this.list).instanceOf(Realm.List);
-
-      this.objectWithDictionary = this.realm.write(() => {
-        return this.realm.create(ObjectWithMixed, { mixedValue: {} });
-      });
-      this.dictionary = this.objectWithDictionary.mixedValue as Realm.Dictionary<any>;
-      expect(this.dictionary).instanceOf(Realm.Dictionary);
+      this.objectWithList = this.realm.write(() => this.realm.create(ObjectWithMixed, { mixed: [] }));
+      this.objectWithDictionary = this.realm.write(() => this.realm.create(ObjectWithMixed, { mixed: {} }));
     });
 
     describe("Collection notifications", () => {
-      it("fires when inserting to top-level list", async function (this: CollectionsInMixedContext) {
-        await expectCollectionNotifications(this.list, undefined, [
-          EMPTY_COLLECTION_CHANGESET,
-          () => {
-            this.realm.write(() => {
-              this.list.push("Amy");
-              this.list.push("Mary");
-              this.list.push("John");
-            });
-          },
-          {
-            deletions: [],
-            insertions: [0, 1, 2],
-            newModifications: [],
-            oldModifications: [],
-          },
-        ]);
-      });
+      describe("List", () => {
+        it("fires when inserting, updating, and deleting at top-level", async function (this: CollectionsInMixedContext) {
+          const list = this.objectWithList.mixed;
+          expectRealmList(list);
 
-      it("fires when inserting to top-level dictionary", async function (this: CollectionsInMixedContext) {
-        await expectDictionaryNotifications(this.dictionary, undefined, [
-          EMPTY_DICTIONARY_CHANGESET,
-          () => {
-            this.realm.write(() => {
-              this.dictionary.amy = "Amy";
-              this.dictionary.mary = "Mary";
-              this.dictionary.john = "John";
-            });
-          },
-          {
-            deletions: [],
-            insertions: ["amy", "mary", "john"],
-            modifications: [],
-          },
-        ]);
-      });
+          await expectCollectionNotifications(list, undefined, [
```

Comment:
It seems to me none of the new tests are using key-path filtering, is that correct / on purpose? I'd think it makes sense to exercise that for nested collections in mixed too.

Reply:
For collection listeners, changes to a Realm Object in the collection will not fire a notification. But for object listeners, we could e.g. test that "Changes to a nested collection fires a notification" (if we're listening on a Realm Object with the key path being the mixed property) 👍 

Currently, the above test exists where the key path is undefined. I think our preexisting key path tests take care of testing that notifications are not fired when non-specified Realm Object properties change.

Did you have any other tests in mind?

Reply:
> For collection listeners, changes to a Realm Object in the collection will not fire a notification.

Is this true when explicitly providing a key-path? 🤔 And is this expected? (I'm unsure)

Reply:
I've tested some of this locally, and currently the key paths option is only supported if the collection is a collection of Realm objects in the schema. So if it's a Mixed in the schema, but the underlying value is dictionary of Realm objects, it's not supported. I believe this may change in Core at a later point.